### ISSUE TITLE
A component is `contentEditable` and contains `children` managed by React in Suneditor while working in Next.js.

### ISSUE THREAD

**Issue by divrawat** [mkhstar/suneditor-react#304]
The images and Iframes that I add in the suneditor are wrapped in class ".se-component" that has contenteditable=" false" attribute within it. Because of this I am getting error or warning in console only after fetching and displaying the post from MongoDB. If I do not use any image or embed video in a blogpost then there is no such error or warning.

![Suneditor2](https://user-images.githubusercontent.com/83118748/227184473-ac0489d1-53d5-4b54-a507-2dfa3b825a0a.png)

If I remove contenteditable attribute with the use of Useeffect in Next.js then the attribute is removed as shown in DOM but the warning or error still stays.

![Suneditor](https://user-images.githubusercontent.com/83118748/227184548-71d4921a-8809-4932-a306-1f658f1bc48e.png)

If I remove contenteditable=" false" from source code while writing a post then the warning or error is gone after fetching and displaying post from MongoDB.

So I completely want to remove contenteditable=" false" attribute from ".se-component".

Please Help .

**Comment by mkhstar** [comment]
@divrawat could you give a demo?

**Comment by mkhstar** [comment]
I mean some demo that demonstrates this issue maybe using codesandbox

**Comment by divrawat** [comment]
> I mean some demo that demonstrates this issue maybe using codesandbox

I can not give Demo since my project is in local machine but -

Its simple, Adding images or iframes through suneditor to a blogpost or article will give warning in console  "A component is contentEditable and contains children managed by React" after viewing that article or blogposts.


![1](https://user-images.githubusercontent.com/83118748/232467099-1a07c0f1-a85a-481c-8baf-eb4d88a6b596.png)



This is because when you add image or iframe then an attribute    **contenteditable=" false"**    is attached to image component or div of SunEditor.  And because of this, warning can be seen in console.
![3](https://user-images.githubusercontent.com/83118748/232467308-20110bf9-6391-4c45-b792-c21788ecf599.png)





If I remove contenteditable=" false" from     **source of Suneditor**   while writing the article    and then view that article after publishing then the warning is gone.



![4](https://user-images.githubusercontent.com/83118748/232467529-500feaf4-78b0-4900-b8f3-e382b067e149.png)




There is no effect of this warning on My project.
But I am not able to suppress this warning by any means.

**Comment by mkhstar** [comment]
@JiHong88 is there a reason why we add `contenteditable=" false"`? I guess that is causing the issue. @divrawat  probably this pr should have been opened in the main repo

**Comment by mkhstar** [comment]
@divrawat do you have a parent that is having a `contenteditable`? That could be causing the issue as well

**Comment by divrawat** [comment]
> @divrawat do you have a parent that is having a `contenteditable`? That could be causing the issue as well

No, the issues are only due to SunEditor's components like Image, Iframes that has  contenteditable=" false" attached to them while uploading.

If I don't upload any Image or Iframe then the console is clean hence No Warnings.

**Comment by JiHong88** [comment]
"contenteditable=false" is used to maintain the format of media tags such as images and iframes.
It looks good to remove "contenteditable=false" when getting content with "getContents".
This will solve the current issue and there will be no problems with the editing function.
I will update it as soon as possible.
